### ISSUE TITLE
How to edit about nullable values?

### ISSUE THREAD

**Issue by kanryu** [telerik/UI-For-UWP#257]
Hi,

I want to prepare GUI for editing ObservableCollection<MyViewModel>.
So I'll use Telerik RadDataGrid and RadDatePicker, but I found problems.

DataGridNumericalColumn cannot edit values of type 'int?'.
DataGridDateColumn cannot show empty when the value is null for type 'DateTime?'.

The area on the screen required by the RadDatePicker is too large to be placed on the screen. The default string "Select Date" is clearly unnecessary. Please display only the current date and the pull down symbol.

Comments?

**Comment by APopatanasov** [comment]
@kanryu  With the current implementation of the NumericBox the control works with double?. Indeed it could be improved to work with int? however, currently it only works with double?. If you need to have a visualization of the values to be like int inside the editor you can use our new approach for building custom DataGrid columns. You can inherit from the DataGridNumericColumn and set a ValueFormat to be {0:N0} for example and no trailing numbers should be visualized.

About the DatePickerColumn you can use the same approach - to build custom column and set the EmptyContent to null. Check the attached sample that demonstrates how to achieve that. 

I will mark this issue as feature request for improving the NumericBox to work with int? and to introduce easy way for customization of the DataGrid editors.

If you are interested on working on these features please, let me know and I will give you some advises from where you can start.
[RadDataGridIssue_257.zip](https://github.com/telerik/UI-For-UWP/files/1840810/RadDataGridIssue_257.zip)

**Comment by kanryu** [comment]
Thanks for you support.

> With the current implementation of the NumericBox the control works with double?

When I experimented, when I display the value on the screen using DataGridNumericalColumn for the property 'int?', The value is displayed correctly on the screen. And I can switch to edit mode. However, when I finalize the edits, the value will return to the value before editing.

**Comment by APopatanasov** [comment]
@kanryu That observation is correct - when finalizing the edit the parser inside the control tries to parse it to double and if the parsing fails the previous value is restored.

Currently, this is expected behavior because as mentioned in my initial reply the control works only with double?. If you are interested on implementing the logic for working with int? please, let me know and I will give you some starts points.

**Comment by kanryu** [comment]
@APopatanasov For business use 'int?' And 'decimal?' Are used much more than 'double?', So please work with these nullables as well.

Likewise, for DataGridDateColumn, please provide a means of customizing the dialog as with RadDatePicker. Perhaps it is possible if we describe the tag of XAML and describe it.

**Comment by APopatanasov** [comment]
@kanryu I agree that the control could be improved and because of that the issue is marked as a feature request for improving it - nullables will be taken into account when the improvements are made.

For the DataGridColumn I forget to mention that there is an easier approach for Styling the column's editor using the CellEditorStyle property - using it you can try to template the control visualized for editing as desired:

`<telerikGrid:DataGridDateColumn PropertyName="NationalHoliday">
    <telerikGrid:DataGridDateColumn.CellEditorStyle>
        <Style TargetType="input:RadDatePicker">
            <Setter Property="EmptyContent" Value=""/>
        </Style>
    </telerikGrid:DataGridDateColumn.CellEditorStyle>
</telerikGrid:DataGridDateColumn>`

Hope the information will be helpful for you.
